src/lib/backoff.ts

```diff
@@ -0,0 +1,66 @@
+import { promisify } from 'util';
+
+const sleep = promisify(setTimeout);
+
+export type Options = {
+	attempt: number;
+	maxDelay: number;
+	minDelay: number;
+	maxRetries: number;
+	retryCallback?: (msg: string) => void;
+};
+
+const DEFAULT_OPTIONS: Partial<Options> = {
+	attempt: 0,
+	maxDelay: 15000,
+	minDelay: 15000,
+	maxRetries: 10,
```

Comment:
Maybe `maxRetries` could default to `+Infinity`, which would make your logic below still work but will have the default behavior that it will retry forever.